tests/test_internals/test_hdmf.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+from hdmf.testing import TestCase
+
+from nwb_conversion_tools.tools.hdmf import SliceableDataChunkIterator
+
+
+class TestIteratorAssertions(TestCase):
+    def test_buffer_bigger_than_chunk_assertion(self):
+        with self.assertRaisesWith(
+            AssertionError, exc_msg="buffer_gb (5e-06) must be greater than the chunk size (0.000996872)!"
+        ):
+            SliceableDataChunkIterator(data=np.random.random(size=(10**3, 10**3)), buffer_gb=0.000005)
+
+
+def test_early_exit():
+    """Uses a 32 byte array with 1 GB buffer size (default) and 1 MB chunk size (default)."""
+    iterator = SliceableDataChunkIterator(data=np.random.random(size=(2, 2)))
+    assert iterator.maxshape == iterator.buffer_shape
+
+
+def test_buffer_padding_long_shape():
+    """Uses ~8 MB array with 11 MB buffer size and 1 MB chunk size (default)."""
+    iterator = SliceableDataChunkIterator(data=np.zeros(shape=(160000, 20)), buffer_gb=1.1e-2)
```

Comment:
Maybe we should use `np.empty` instead of `np.zeros`. Not sure what will happen under the hood (of the DataChunkIterator` but if it works well it should be faster.